
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  
  <!-- Licensed under the Apache 2.0 License -->
  <link rel="stylesheet" type="text/css" href="_static/fonts/open-sans/stylesheet.css" />
  <!-- Licensed under the SIL Open Font License -->
  <link rel="stylesheet" type="text/css" href="_static/fonts/source-serif-pro/source-serif-pro.css" />
  <link rel="stylesheet" type="text/css" href="_static/css/bootstrap.min.css" />
  <link rel="stylesheet" type="text/css" href="_static/css/bootstrap-theme.min.css" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
    <title>pysradb &#8212; pysradb 0.6.0 documentation</title>
    <link rel="stylesheet" href="_static/guzzle.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/graphviz.css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/language_data.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Installation" href="installation.html" />
    <link rel="prev" title="Welcome to pysradb’s documentation!" href="index.html" />
  
   

  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="installation.html" title="Installation"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="index.html" title="Welcome to pysradb’s documentation!"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">pysradb 0.6.0 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="container-wrapper">

      <div id="mobile-toggle">
        <a href="#"><span class="glyphicon glyphicon-align-justify" aria-hidden="true"></span></a>
      </div>
  <div id="left-column">
    <div class="sphinxsidebar">
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <h2>Contents</h2>
    <div class="sidebar-localtoc">
      <p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">pysradb</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#demo-notebooks">Demo Notebooks</a></li>
<li class="toctree-l2"><a class="reference internal" href="#installation">Installation</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#dependecies">Dependecies</a></li>
<li class="toctree-l3"><a class="reference internal" href="#downloading-srametadb">Downloading SRAmetadb</a></li>
<li class="toctree-l3"><a class="reference internal" href="#aspera-client">aspera-client</a></li>
<li class="toctree-l3"><a class="reference internal" href="#installing-pysradb-in-development-mode">Installing pysradb in development mode</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#interacting-with-sra">Interacting with SRA</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#use-case-1-fetch-the-metadata-table-sra-runtable">Use Case 1: Fetch the metadata table (SRA-runtable)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#use-case-2-downloading-an-entire-project-arranged-experiment-wise">Use Case 2: Downloading an entire project arranged experiment wise</a></li>
<li class="toctree-l3"><a class="reference internal" href="#use-case-3-downloading-a-subset-of-experiments">Use Case 3: Downloading a subset of experiments</a></li>
<li class="toctree-l3"><a class="reference internal" href="#use-case-4-getting-cell-type-treatment-information-from-sample-attributes">Use Case 4: Getting cell-type/treatment information from sample_attributes</a></li>
<li class="toctree-l3"><a class="reference internal" href="#use-case-5-searching-for-datasets">Use Case 5: Searching for datasets</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#citation">Citation</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="cmdline.html">CLI usage</a></li>
<li class="toctree-l1"><a class="reference internal" href="usage.html">Usage</a></li>
<li class="toctree-l1"><a class="reference internal" href="modules.html">pysradb</a></li>
<li class="toctree-l1"><a class="reference internal" href="contributing.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="authors.html">Credits</a></li>
<li class="toctree-l1"><a class="reference internal" href="history.html">History</a></li>
</ul>

    </div>
  </div>
</div>
  <h4>Previous topic</h4>
  <p class="topless"><a href="index.html"
                        title="previous chapter">Welcome to pysradb’s documentation!</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="installation.html"
                        title="next chapter">Installation</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/readme.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <div id="main-search">
      <form class="form-inline" action="search.html" method="GET" role="form">
        <div class="input-group">
          <input name="q" type="text" class="form-control" placeholder="Search...">
        </div>
        <input type="hidden" name="check_keywords" value="yes" />
        <input type="hidden" name="area" value="default" />
      </form>
    </div>
  </div>
</div>
      
    </div>
  </div>
        <div id="right-column">
          
          <div role="navigation" aria-label="breadcrumbs navigation">
            <ol class="breadcrumb">
              <li><a href="index.html">Docs</a></li>
              
              <li>pysradb</li>
            </ol>
          </div>
          
          <div class="document clearer body">
            
  <div class="section" id="pysradb">
<h1>pysradb<a class="headerlink" href="#pysradb" title="Permalink to this headline">¶</a></h1>
<a class="reference external image-reference" href="https://zenodo.org/badge/latestdoi/159590788"><img alt="https://zenodo.org/badge/159590788.svg" src="https://zenodo.org/badge/159590788.svg" /></a>
<a class="reference external image-reference" href="https://pypi.python.org/pypi/pysradb"><img alt="https://img.shields.io/pypi/v/pysradb.svg?style=flat-square" src="https://img.shields.io/pypi/v/pysradb.svg?style=flat-square" /></a>
<a class="reference external image-reference" href="https://travis-ci.com/saketkc/pysradb"><img alt="https://img.shields.io/travis-ci/saketkc/pysradb.svg?style=flat-square" src="https://img.shields.io/travis-ci/saketkc/pysradb.svg?style=flat-square" /></a>
<a class="reference external image-reference" href="http://bioconda.github.io/recipes/pysradb/README.html"><img alt="https://img.shields.io/badge/install%20with-bioconda-brightgreen.svg?style=flat-square" src="https://img.shields.io/badge/install%20with-bioconda-brightgreen.svg?style=flat-square" /></a>
<a class="reference external image-reference" href="https://codecov.io/gh/saketkc/pysradb"><img alt="https://codecov.io/gh/saketkc/pysradb/branch/master/graph/badge.svg?style=flat-square" src="https://codecov.io/gh/saketkc/pysradb/branch/master/graph/badge.svg?style=flat-square" /></a>
<p>Python package for interacting with SRAdb and downloading datasets from SRA.</p>
<a href="https://asciinema.org/a/0C3SjYmPTkkemldprUpdVhiKx?speed=5&autoplay=1" target="_blank"><img src="https://asciinema.org/a/0C3SjYmPTkkemldprUpdVhiKx.svg" /></a><div class="section" id="demo-notebooks">
<h2>Demo Notebooks<a class="headerlink" href="#demo-notebooks" title="Permalink to this headline">¶</a></h2>
<p>These notebooks document all the possible features of <cite>pysradb</cite>:</p>
<ol class="arabic simple">
<li><a class="reference external" href="https://nbviewer.jupyter.org/github/saketkc/pysradb/blob/master/notebooks/01.SRAdb-demo.ipynb">SRAmetadb operations</a></li>
<li><a class="reference external" href="https://nbviewer.jupyter.org/github/saketkc/pysradb/blob/master/notebooks/02.GEOmetadb-demo.ipynb">GEOmetadb operations</a></li>
<li><a class="reference external" href="https://nbviewer.jupyter.org/github/saketkc/pysradb/blob/master/notebooks/03.CommandLine-demo.ipynb">Command line usage</a></li>
</ol>
</div>
<div class="section" id="installation">
<h2>Installation<a class="headerlink" href="#installation" title="Permalink to this headline">¶</a></h2>
<p>To install stable version using <cite>pip</cite>:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pip install pysradb
</pre></div>
</div>
<p>Alternatively, if you use conda:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>conda install -c bioconda pysradb
</pre></div>
</div>
<p>This step will install all the dependencies except <a class="reference internal" href="#aspera-client">aspera-client</a> (which is not required, but highly recommended).
Both Python 2 and Python 3 are supported.</p>
<div class="section" id="dependecies">
<h3>Dependecies<a class="headerlink" href="#dependecies" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pandas&gt;<span class="o">=</span><span class="m">0</span>.23.4
tqdm&gt;<span class="o">=</span><span class="m">4</span>.28
click&gt;<span class="o">=</span><span class="m">7</span>.0
aspera-client
SRAmetadb.sqlite
</pre></div>
</div>
</div>
<div class="section" id="downloading-srametadb">
<h3>Downloading SRAmetadb<a class="headerlink" href="#downloading-srametadb" title="Permalink to this headline">¶</a></h3>
<p>We need a SQLite database file that has preprocessed metadata made available by the
<a class="reference external" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-19">SRAdb</a> project.</p>
<p>SRAmetadb can be downloaded using:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>wget -c https://starbuck1.s3.amazonaws.com/sradb/SRAmetadb.sqlite.gz <span class="o">&amp;&amp;</span> gunzip SRAmetadb.sqlite.gz
</pre></div>
</div>
<p>Alternatively, you can also download it using <cite>pysradb</cite>:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pysradb</span> <span class="kn">import</span> <span class="n">download_sradb_file</span>
<span class="n">download_sradb_file</span><span class="p">()</span>

<span class="n">SRAmetadb</span><span class="o">.</span><span class="n">sqlite</span><span class="o">.</span><span class="n">gz</span><span class="p">:</span> <span class="mf">2.44</span><span class="n">GB</span> <span class="p">[</span><span class="mo">01</span><span class="p">:</span><span class="mi">10</span><span class="p">,</span> <span class="mf">36.9</span><span class="n">MB</span><span class="o">/</span><span class="n">s</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="section" id="aspera-client">
<span id="id1"></span><h3>aspera-client<a class="headerlink" href="#aspera-client" title="Permalink to this headline">¶</a></h3>
<p>We strongly recommend using <cite>aspera-client</cite> (which uses UDP) since it <a class="reference external" href="http://www.skullbox.net/tcpudp.php">warrants faster downloads</a> as compared to <cite>ftp/http</cite> based downloads.</p>
<p>PDF intructions are available on IBM’s <a class="reference external" href="https://downloads.asperasoft.com/connect2/">website</a>.</p>
<p>Direct download links:</p>
<ul class="simple">
<li>Linux: <a class="reference external" href="https://download.asperasoft.com/download/sw/connect/3.8.1/ibm-aspera-connect-3.8.1.161274-linux-g2.12-64.tar.gz">https://download.asperasoft.com/download/sw/connect/3.8.1/ibm-aspera-connect-3.8.1.161274-linux-g2.12-64.tar.gz</a></li>
<li>MacOS: <a class="reference external" href="https://download.asperasoft.com/download/sw/connect/3.8.1/IBMAsperaConnectInstaller-3.8.1.161274.dmg">https://download.asperasoft.com/download/sw/connect/3.8.1/IBMAsperaConnectInstaller-3.8.1.161274.dmg</a></li>
<li>Windows: <a class="reference external" href="https://download.asperasoft.com/download/sw/connect/3.8.1/IBMAsperaConnect-ML-3.8.1.161274.msi">https://download.asperasoft.com/download/sw/connect/3.8.1/IBMAsperaConnect-ML-3.8.1.161274.msi</a></li>
</ul>
<p>Once you download the tar relevant to your OS, say linux, follow these steps to install aspera:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tar -zxvf ibm-aspera-connect-3.8.1.161274-linux-g2.12-64.tar.gz
bash ibm-aspera-connect-3.8.1.161274-linux-g2.12-64.sh
Installing IBM Aspera Connect
Deploying IBM Aspera Connect <span class="o">(</span>/home/saket/.aspera/connect<span class="o">)</span> <span class="k">for</span> the current user only.
Install complete.
</pre></div>
</div>
</div>
<div class="section" id="installing-pysradb-in-development-mode">
<h3>Installing pysradb in development mode<a class="headerlink" href="#installing-pysradb-in-development-mode" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pip install -U pandas tqdm
git clone https://github.com/saketkc/pysradb.git
<span class="nb">cd</span> pysradb
pip install -e .
</pre></div>
</div>
</div>
</div>
<div class="section" id="interacting-with-sra">
<h2>Interacting with SRA<a class="headerlink" href="#interacting-with-sra" title="Permalink to this headline">¶</a></h2>
<div class="section" id="use-case-1-fetch-the-metadata-table-sra-runtable">
<h3>Use Case 1: Fetch the metadata table (SRA-runtable)<a class="headerlink" href="#use-case-1-fetch-the-metadata-table-sra-runtable" title="Permalink to this headline">¶</a></h3>
<p>The simplest use case of <cite>pysradb</cite> is when you apriopri know the SRA project ID (SRP)
and would simply want to fetch the metadata associated with it. This is generally
reflected in the <cite>SraRunTable.txt</cite> that you get from NCBI’s website.
See an <a class="reference external" href="https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP098789">example</a> of a SraRunTable.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pysradb</span> <span class="kn">import</span> <span class="n">SRAdb</span>
<span class="n">db</span> <span class="o">=</span> <span class="n">SRAdb</span><span class="p">(</span><span class="s1">&#39;SRAmetadb.sqlite&#39;</span><span class="p">)</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">db</span><span class="o">.</span><span class="n">sra_metadata</span><span class="p">(</span><span class="s1">&#39;SRP098789&#39;</span><span class="p">)</span>
<span class="n">df</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
<table border="1" class="docutils">
<colgroup>
<col width="6%" />
<col width="8%" />
<col width="29%" />
<col width="5%" />
<col width="3%" />
<col width="7%" />
<col width="6%" />
<col width="7%" />
<col width="6%" />
<col width="5%" />
<col width="4%" />
<col width="3%" />
<col width="5%" />
<col width="6%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">study_accession</th>
<th class="head">experiment_accession</th>
<th class="head">experiment_title</th>
<th class="head">run_accession</th>
<th class="head">taxon_id</th>
<th class="head">library_selection</th>
<th class="head">library_layout</th>
<th class="head">library_strategy</th>
<th class="head">library_source</th>
<th class="head">library_name</th>
<th class="head">bases</th>
<th class="head">spots</th>
<th class="head">adapter_spec</th>
<th class="head">avg_read_length</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>SRP098789</td>
<td>SRX2536403</td>
<td>GSM2475997: 1.5 Ã‚ÂµM PF-067446846, 10 min, rep 1; Homo sapiens; OTHER</td>
<td>SRR5227288</td>
<td>9606</td>
<td>other</td>
<td>SINGLE -</td>
<td>OTHER</td>
<td>TRANSCRIPTOMIC</td>
<td>&#160;</td>
<td>2104142750</td>
<td>42082855</td>
<td>&#160;</td>
<td>50</td>
</tr>
<tr class="row-odd"><td>SRP098789</td>
<td>SRX2536404</td>
<td>GSM2475998: 1.5 Ã‚ÂµM PF-067446846, 10 min, rep 2; Homo sapiens; OTHER</td>
<td>SRR5227289</td>
<td>9606</td>
<td>other</td>
<td>SINGLE -</td>
<td>OTHER</td>
<td>TRANSCRIPTOMIC</td>
<td>&#160;</td>
<td>2082873050</td>
<td>41657461</td>
<td>&#160;</td>
<td>50</td>
</tr>
<tr class="row-even"><td>SRP098789</td>
<td>SRX2536405</td>
<td>GSM2475999: 1.5 Ã‚ÂµM PF-067446846, 10 min, rep 3; Homo sapiens; OTHER</td>
<td>SRR5227290</td>
<td>9606</td>
<td>other</td>
<td>SINGLE -</td>
<td>OTHER</td>
<td>TRANSCRIPTOMIC</td>
<td>&#160;</td>
<td>2023148650</td>
<td>40462973</td>
<td>&#160;</td>
<td>50</td>
</tr>
<tr class="row-odd"><td>SRP098789</td>
<td>SRX2536406</td>
<td>GSM2476000: 0.3 Ã‚ÂµM PF-067446846, 10 min, rep 1; Homo sapiens; OTHER</td>
<td>SRR5227291</td>
<td>9606</td>
<td>other</td>
<td>SINGLE -</td>
<td>OTHER</td>
<td>TRANSCRIPTOMIC</td>
<td>&#160;</td>
<td>2057165950</td>
<td>41143319</td>
<td>&#160;</td>
<td>50</td>
</tr>
<tr class="row-even"><td>SRP098789</td>
<td>SRX2536407</td>
<td>GSM2476001: 0.3 Ã‚ÂµM PF-067446846, 10 min, rep 2; Homo sapiens; OTHER</td>
<td>SRR5227292</td>
<td>9606</td>
<td>other</td>
<td>SINGLE -</td>
<td>OTHER</td>
<td>TRANSCRIPTOMIC</td>
<td>&#160;</td>
<td>3027621850</td>
<td>60552437</td>
<td>&#160;</td>
<td>50</td>
</tr>
</tbody>
</table>
<p>The metadata is returned as a <cite>pandas</cite> dataframe and hence allows you to perform
all regular select/query operations available through <cite>pandas</cite>.</p>
</div>
<div class="section" id="use-case-2-downloading-an-entire-project-arranged-experiment-wise">
<h3>Use Case 2: Downloading an entire project arranged experiment wise<a class="headerlink" href="#use-case-2-downloading-an-entire-project-arranged-experiment-wise" title="Permalink to this headline">¶</a></h3>
<p>Once you have fetched the metadata and made sure, this is the project
you were looking for, you would want to download everything at once.
NCBI follows this hiererachy: <cite>SRP =&gt; SRX =&gt; SRR</cite>. Each <cite>SRP</cite> (project) has multiple
<cite>SRX</cite> (experiments) and each <cite>SRX</cite> in turn has multiple <cite>SRR</cite> (runs) inside it.
We want to mimick this hiereachy in our downloads. The reason to do that is simple:
in most cases you care about <cite>SRX</cite> the most, and would want to “merge” your SRRs
in one way or the other. Having this hierearchy ensures your downstream code
can handle such cases easily, without worrying about which runs (SRR) need to be merged.</p>
<p>We strongly recommend installing <cite>aspera-client</cite> which uses UDP and is <a class="reference external" href="http://www.skullbox.net/tcpudp.php">designed to be faster</a>.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pysradb</span> <span class="kn">import</span> <span class="n">SRAdb</span>
<span class="n">db</span> <span class="o">=</span> <span class="n">SRAdb</span><span class="p">(</span><span class="s1">&#39;SRAmetadb.sqlite&#39;</span><span class="p">)</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">db</span><span class="o">.</span><span class="n">sra_metadata</span><span class="p">(</span><span class="s1">&#39;SRP017942&#39;</span><span class="p">)</span>
<span class="n">db</span><span class="o">.</span><span class="n">download</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
</pre></div>
</div>
<p>The default download location is <cite>pysradb_downloads/</cite> created inside your current working directory.
You can specify a location by:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">db</span><span class="o">.</span><span class="n">download</span><span class="p">(</span><span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span> <span class="n">out_dir</span><span class="o">=</span><span class="s1">&#39;/pysradb_downloads&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="use-case-3-downloading-a-subset-of-experiments">
<h3>Use Case 3: Downloading a subset of experiments<a class="headerlink" href="#use-case-3-downloading-a-subset-of-experiments" title="Permalink to this headline">¶</a></h3>
<p>Often, you need to process only a smaller set of samples from a project (SRP).
Consider this project which has data spanning four assays.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">df</span> <span class="o">=</span> <span class="n">db</span><span class="o">.</span><span class="n">sra_metadata</span><span class="p">(</span><span class="s1">&#39;SRP000941&#39;</span><span class="p">)</span>
<span class="k">print</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">library_strategy</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span>
<span class="p">[</span><span class="s1">&#39;ChIP-Seq&#39;</span> <span class="s1">&#39;Bisulfite-Seq&#39;</span> <span class="s1">&#39;RNA-Seq&#39;</span> <span class="s1">&#39;WGS&#39;</span> <span class="s1">&#39;OTHER&#39;</span><span class="p">]</span>
</pre></div>
</div>
<p>But, you might be only interested in analyzing the <cite>RNA-seq</cite> samples and would just want to download that subset.
This is simple using <cite>pysradb</cite> since the metadata can be subset just as you would subset a dataframe in
pandas.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">df_rna</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="o">.</span><span class="n">library_strategy</span> <span class="o">==</span> <span class="s1">&#39;RNA-Seq&#39;</span><span class="p">]</span>
<span class="n">db</span><span class="o">.</span><span class="n">download</span><span class="p">(</span><span class="n">df</span><span class="o">=</span><span class="n">df_rna</span><span class="p">,</span> <span class="n">out_dir</span><span class="o">=</span><span class="s1">&#39;/pysradb_downloads&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="use-case-4-getting-cell-type-treatment-information-from-sample-attributes">
<h3>Use Case 4: Getting cell-type/treatment information from sample_attributes<a class="headerlink" href="#use-case-4-getting-cell-type-treatment-information-from-sample-attributes" title="Permalink to this headline">¶</a></h3>
<p>Cell type/tissue informations is usually hidden in the <cite>sample_attributes</cite> column,
which can be expanded:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pysradb.filter_attrs</span> <span class="kn">import</span> <span class="n">expand_sample_attribute_columns</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">db</span><span class="o">.</span><span class="n">sra_metadata</span><span class="p">(</span><span class="s1">&#39;SRP017942&#39;</span><span class="p">)</span>
<span class="n">expand_sample_attribute_columns</span><span class="p">(</span><span class="n">df</span><span class="p">)</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
<table border="1" class="docutils">
<colgroup>
<col width="3%" />
<col width="4%" />
<col width="14%" />
<col width="5%" />
<col width="30%" />
<col width="3%" />
<col width="2%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="3%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="3%" />
<col width="6%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">study_accession</th>
<th class="head">experiment_accession</th>
<th class="head">experiment_title</th>
<th class="head">experiment_attribute</th>
<th class="head">sample_attribute</th>
<th class="head">run_accession</th>
<th class="head">taxon_id</th>
<th class="head">library_selection</th>
<th class="head">library_layout</th>
<th class="head">library_strategy</th>
<th class="head">library_source</th>
<th class="head">library_name</th>
<th class="head">bases</th>
<th class="head">spots</th>
<th class="head">adapter_spec</th>
<th class="head">avg_read_length</th>
<th class="head">assay_type</th>
<th class="head">cell_line</th>
<th class="head">source_name</th>
<th class="head">transfected_with</th>
<th class="head">treatment</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>SRP017942</td>
<td>SRX217028</td>
<td>GSM1063575: 293T_GFP; Homo sapiens; RNA-Seq</td>
<td>GEO Accession: GSM1063575</td>
<td>source_name: 293T cells || cell line: 293T cells || transfected with: 3XFLAG-GFP || assay type: Riboseq</td>
<td>SRR648667</td>
<td>9606</td>
<td>other</td>
<td>SINGLE -</td>
<td>RNA-Seq</td>
<td>TRANSCRIPTOMIC</td>
<td>&#160;</td>
<td>1806641316</td>
<td>50184481</td>
<td>&#160;</td>
<td>36</td>
<td>riboseq</td>
<td>293t cells</td>
<td>293t cells</td>
<td>3xflag-gfp</td>
<td>NaN</td>
</tr>
<tr class="row-odd"><td>SRP017942</td>
<td>SRX217029</td>
<td>GSM1063576: 293T_GFP_2hrs_severe_Heat_Shock; Homo sapiens; RNA-Seq</td>
<td>GEO Accession: GSM1063576</td>
<td>source_name: 293T cells || cell line: 293T cells || transfected with: 3XFLAG-GFP || treatment: severe heat shock (44C 2 hours) || assay type: Riboseq</td>
<td>SRR648668</td>
<td>9606</td>
<td>other</td>
<td>SINGLE -</td>
<td>RNA-Seq</td>
<td>TRANSCRIPTOMIC</td>
<td>&#160;</td>
<td>3436984836</td>
<td>95471801</td>
<td>&#160;</td>
<td>36</td>
<td>riboseq</td>
<td>293t cells</td>
<td>293t cells</td>
<td>3xflag-gfp</td>
<td>severe heat shock (44c 2 hours)</td>
</tr>
<tr class="row-even"><td>SRP017942</td>
<td>SRX217030</td>
<td>GSM1063577: 293T_Hspa1a; Homo sapiens; RNA-Seq</td>
<td>GEO Accession: GSM1063577</td>
<td>source_name: 293T cells || cell line: 293T cells || transfected with: 3XFLAG-Hspa1a || assay type: Riboseq</td>
<td>SRR648669</td>
<td>9606</td>
<td>other</td>
<td>SINGLE -</td>
<td>RNA-Seq</td>
<td>TRANSCRIPTOMIC</td>
<td>&#160;</td>
<td>3330909216</td>
<td>92525256</td>
<td>&#160;</td>
<td>36</td>
<td>riboseq</td>
<td>293t cells</td>
<td>293t cells</td>
<td>3xflag-hspa1a</td>
<td>NaN</td>
</tr>
<tr class="row-odd"><td>SRP017942</td>
<td>SRX217031</td>
<td>GSM1063578: 293T_Hspa1a_2hrs_severe_Heat_Shock; Homo sapiens; RNA-Seq</td>
<td>GEO Accession: GSM1063578</td>
<td>source_name: 293T cells || cell line: 293T cells || transfected with: 3XFLAG-Hspa1a || treatment: severe heat shock (44C 2 hours) || assay type: Riboseq</td>
<td>SRR648670</td>
<td>9606</td>
<td>other</td>
<td>SINGLE -</td>
<td>RNA-Seq</td>
<td>TRANSCRIPTOMIC</td>
<td>&#160;</td>
<td>3622123512</td>
<td>100614542</td>
<td>&#160;</td>
<td>36</td>
<td>riboseq</td>
<td>293t cells</td>
<td>293t cells</td>
<td>3xflag-hspa1a</td>
<td>severe heat shock (44c 2 hours)</td>
</tr>
<tr class="row-even"><td>SRP017942</td>
<td>SRX217956</td>
<td>GSM794854: 3T3-Control-Riboseq; Mus musculus; RNA-Seq</td>
<td>GEO Accession: GSM794854</td>
<td>source_name: 3T3 cells || treatment: control || cell line: 3T3 cells || assay type: Riboseq</td>
<td>SRR649752</td>
<td>10090</td>
<td>cDNA</td>
<td>SINGLE -</td>
<td>RNA-Seq</td>
<td>TRANSCRIPTOMIC</td>
<td>&#160;</td>
<td>594945396</td>
<td>16526261</td>
<td>&#160;</td>
<td>36</td>
<td>riboseq</td>
<td>3t3 cells</td>
<td>3t3 cells</td>
<td>NaN</td>
<td>control</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="use-case-5-searching-for-datasets">
<h3>Use Case 5: Searching for datasets<a class="headerlink" href="#use-case-5-searching-for-datasets" title="Permalink to this headline">¶</a></h3>
<p>Another common operation that we do on SRA is seach, plain text search.</p>
<p>If you want to look up for all projects where <cite>ribosome profiling</cite> appears somewhere
in the description:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">df</span> <span class="o">=</span> <span class="n">db</span><span class="o">.</span><span class="n">search_sra</span><span class="p">(</span><span class="n">search_str</span><span class="o">=</span><span class="s1">&#39;&quot;ribosome profiling&quot;&#39;</span><span class="p">)</span>
<span class="n">df</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
<table border="1" class="docutils">
<colgroup>
<col width="7%" />
<col width="10%" />
<col width="27%" />
<col width="6%" />
<col width="4%" />
<col width="8%" />
<col width="7%" />
<col width="8%" />
<col width="7%" />
<col width="6%" />
<col width="5%" />
<col width="4%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">study_accession</th>
<th class="head">experiment_accession</th>
<th class="head">experiment_title</th>
<th class="head">run_accession</th>
<th class="head">taxon_id</th>
<th class="head">library_selection</th>
<th class="head">library_layout</th>
<th class="head">library_strategy</th>
<th class="head">library_source</th>
<th class="head">library_name</th>
<th class="head">bases</th>
<th class="head">spots</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>DRP003075</td>
<td>DRX019536</td>
<td>Illumina Genome Analyzer IIx sequencing of SAMD00018584</td>
<td>DRR021383</td>
<td>83333</td>
<td>other</td>
<td>SINGLE -</td>
<td>OTHER</td>
<td>TRANSCRIPTOMIC</td>
<td>GAII05_3</td>
<td>978776480</td>
<td>12234706</td>
</tr>
<tr class="row-odd"><td>DRP003075</td>
<td>DRX019537</td>
<td>Illumina Genome Analyzer IIx sequencing of SAMD00018585</td>
<td>DRR021384</td>
<td>83333</td>
<td>other</td>
<td>SINGLE -</td>
<td>OTHER</td>
<td>TRANSCRIPTOMIC</td>
<td>GAII05_4</td>
<td>894201680</td>
<td>11177521</td>
</tr>
<tr class="row-even"><td>DRP003075</td>
<td>DRX019538</td>
<td>Illumina Genome Analyzer IIx sequencing of SAMD00018586</td>
<td>DRR021385</td>
<td>83333</td>
<td>other</td>
<td>SINGLE -</td>
<td>OTHER</td>
<td>TRANSCRIPTOMIC</td>
<td>GAII05_5</td>
<td>931536720</td>
<td>11644209</td>
</tr>
<tr class="row-odd"><td>DRP003075</td>
<td>DRX019540</td>
<td>Illumina Genome Analyzer IIx sequencing of SAMD00018588</td>
<td>DRR021387</td>
<td>83333</td>
<td>other</td>
<td>SINGLE -</td>
<td>OTHER</td>
<td>TRANSCRIPTOMIC</td>
<td>GAII07_4</td>
<td>2759398700</td>
<td>27593987</td>
</tr>
<tr class="row-even"><td>DRP003075</td>
<td>DRX019541</td>
<td>Illumina Genome Analyzer IIx sequencing of SAMD00018589</td>
<td>DRR021388</td>
<td>83333</td>
<td>other</td>
<td>SINGLE -</td>
<td>OTHER</td>
<td>TRANSCRIPTOMIC</td>
<td>GAII07_5</td>
<td>2386196500</td>
<td>23861965</td>
</tr>
</tbody>
</table>
<p>Again, the results are available as a <cite>pandas</cite> dataframe and hence
you can perform all subset operations post your query. Your query doesn’t need
to be exact.</p>
</div>
</div>
<div class="section" id="citation">
<h2>Citation<a class="headerlink" href="#citation" title="Permalink to this headline">¶</a></h2>
<p>Zenodo archive: <a class="reference external" href="https://zenodo.org/badge/latestdoi/159590788">https://zenodo.org/badge/latestdoi/159590788</a></p>
<p>DOI: 10.5281/zenodo.2306881</p>
<p>A lot of functionality in <cite>pysradb</cite> is based on ideas from the original <a class="reference external" href="https://bioconductor.org/packages/release/bioc/html/SRAdb.html">SRAdb package</a>. Please cite the original SRAdb publication:</p>
<blockquote>
<div>Zhu, Yuelin, Robert M. Stephens, Paul S. Meltzer, and Sean R. Davis. “SRAdb: query and use public next-generation sequencing data from within R.” BMC bioinformatics 14, no. 1 (2013): 19.</div></blockquote>
<ul class="simple">
<li>Free software: BSD license</li>
<li>Documentation: <a class="reference external" href="https://saketkc.github.io/pysradb">https://saketkc.github.io/pysradb</a></li>
</ul>
</div>
</div>


          </div>
            
  <div class="footer-relations">
    
      <div class="pull-left">
        <a class="btn btn-default" href="index.html" title="previous chapter (use the left arrow)">Welcome to pysradb’s documentation!</a>
      </div>
    
      <div class="pull-right">
        <a class="btn btn-default" href="installation.html" title="next chapter (use the right arrow)">Installation</a>
      </div>
    </div>
    <div class="clearer"></div>
  
        </div>
        <div class="clearfix"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="installation.html" title="Installation"
             >next</a> |</li>
        <li class="right" >
          <a href="index.html" title="Welcome to pysradb’s documentation!"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">pysradb 0.6.0 documentation</a> &#187;</li> 
      </ul>
    </div>
<script type="text/javascript">
  $("#mobile-toggle a").click(function () {
    $("#left-column").toggle();
  });
</script>
<script type="text/javascript" src="_static/js/bootstrap.js"></script>
  <div class="footer">
    &copy; Copyright 2018, Saket Choudhary. Created using <a href="http://sphinx.pocoo.org/">Sphinx</a>.
  </div>
  </body>
</html>